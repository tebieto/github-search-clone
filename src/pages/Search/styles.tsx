import styled from 'styled-components';

export const SearchContainer = styled.div`
  .search-header {
    display: flex;
    align-items: center;
    justify-content: flex-end;
    margin: 20px 200px;
    @media screen and (max-width: 800px) {
      margin: 20px;
    }
  }
  .search-body {
    display: flex;
    justify-content: center;
    .search-body-contents {
      display: flex;
      flex-direction: column;
      align-items: center;
      margin: 100px 0;
      button {
        height: 40px !important;
        padding: 0 30px !important;
      }
      .search-input {
        width: 600px;
        margin: 30px 0;
        max-width: 90vw;
      }
    }
  }
`;
