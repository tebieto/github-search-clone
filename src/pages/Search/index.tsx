import React from 'react';
import { useHistory } from 'react-router';
import CustomButton from '../../components/CustomButton';
import CustomInput from '../../components/CustomInput';
import HeaderProfileSection from '../../components/HeaderProfileSection';
import Logo from '../../components/Logo';
import { queryLink } from '../../utils';
import { SearchContainer } from './styles';

const Search = (): JSX.Element => {
  const [inputValue, setInputValue] = React.useState('react');

  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    setInputValue(e.target.value);
  };

  const history = useHistory();

  const handleKeyDown = React.useCallback(
    (e: React.KeyboardEvent<HTMLDivElement>) => {
      e.key === 'Enter' && history.push(queryLink(inputValue));
    },
    [inputValue],
  );

  return (
    <SearchContainer>
      <div className="search-header">
        <HeaderProfileSection />
      </div>
      <div className="search-body">
        <div className="search-body-contents">
          <Logo />
          <div className="search-input">
            <CustomInput
              aria-label="Search Input"
              value={inputValue}
              onChange={handleChange}
              onKeyDown={handleKeyDown}
            />
          </div>
          <CustomButton
            aria-label="Search Github Button"
            onClick={() => history.push(queryLink(inputValue))}
          >
            Search Github
          </CustomButton>
        </div>
      </div>
    </SearchContainer>
  );
};

export default Search;
