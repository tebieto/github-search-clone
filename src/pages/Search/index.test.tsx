import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Search from '.';

it('renders correctly', () => {
  const tree = shallow(<Search />);
  expect(toJson(tree)).toMatchSnapshot();
  expect(tree.contains('Search Github')).toEqual(true);
});
