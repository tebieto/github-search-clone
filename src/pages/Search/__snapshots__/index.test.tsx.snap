// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<styled.div>
  <Title
    title="Github Search | Search"
  />
  <div
    className="search-header"
  >
    <HeaderProfileSection />
  </div>
  <div
    className="search-body"
  >
    <div
      className="search-body-contents"
    >
      <Logo />
      <div
        className="search-input"
      >
        <CustomInput
          aria-label="Search Input"
          onChange={[Function]}
          onKeyDown={[Function]}
          value="react"
        />
      </div>
      <CustomButton
        aria-label="Search Github Button"
        onClick={[Function]}
      >
        Search Github
      </CustomButton>
    </div>
  </div>
</styled.div>
`;
